app/src/main/java/com/example/networkingapp/repository/MainRepository.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.networkingapp.repository
+
+import com.example.networkingapp.userAPI.RetrofitInstance
+import com.example.networkingapp.userAPI.model.RandomUserAPIResponse
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.io.IOException
+
+class MainRepository(private val resultProcessor: ProcessRequestResult) {
```

Comment:
Create inteerface for the `MainRepository` to be able to support DI(dependency inversion)